Set-PathVariable.ps1

```diff
@@ -0,0 +1,80 @@
+﻿[CmdletBinding(SupportsShouldProcess=$true)]
+param(
+    [parameter(Mandatory=$true)]
+    [string]$NewLocation)
+
+Begin
+{
+
+#requires –runasadministrator
+
+    $regPath = "SYSTEM\CurrentControlSet\Control\Session Manager\Environment"
+    $hklm = [Microsoft.Win32.Registry]::LocalMachine
+
+    Function GetOldPath()
+    {     
+        $regKey = $hklm.OpenSubKey($regPath, $FALSE)
+        $envpath = $regKey.GetValue("Path", "", [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames)
+        return $envPath
+    }
+}
+
+Process
+{
+    # Win32API error codes
+    $ERROR_SUCCESS = 0
+    $ERROR_DUP_NAME = 34 
+    $ERROR_INVALID_DATA = 13
+
+    $NewLocation = $NewLocation.Trim();
+
+    If ($NewLocation -eq "" -or $NewLocation -eq $null)
+    {
+        Exit $ERROR_INVALID_DATA
```

Comment:
exit is bad practice because calling code has no chance to clean up. Better to throw or simply return.